eclipse-emfcloud/emfcloud-modelserver-theia
# #179 Consistently use URI object in APIs

Consistently use URI object in APIs

- Use `urijs` as typed URI library
- Use URI objects in APIs and internally for proper URI handling
  - Adapt interfaces, tests
- Customize JsonRpcProxyFactory to ensure URI object is correctly typed onRequest/onNotifciation
  - Configure on backend and frontend for the JsonRpc WebSocketConnection
- Rename modeluri to lowercase where applicable for better alignment
- Differentiate URI and TheiaURI object
- Update README examples

Update dev-example to ModelServerClient V2

- Update custom DevModelServerClient
- Update api-test-menu
  - User URIs to trigger commands
  - Refine notification printing
  - Add test commands that edit via JSON patch commands
  - Remove xmi test commands as they are not relevant anymore
 
General
- Remove simple-import-sort eslint plugin as it interferes with the built-in import sort
- Ignore zipped rolling log files
- Fix launch configs

Part of https://github.com/eclipse-emfcloud/emfcloud/issues/179

Contributed on behalf of STMicroelectronics
